runtime/src/bank.rs

```diff
@@ -4098,39 +4098,25 @@ impl Bank {
         txs: &[SanitizedTransaction],
         execution_results: &[TransactionExecutionResult],
     ) -> Vec<Result<()>> {
-        let hash_queue = self.blockhash_queue.read().unwrap();
```

Comment:
we moved this lock to be once per tx instead of once per batch. I think we should keep it once per batch to avoid unneccessary overhead.
Can we pass the hash_queue reference to `get_details_from_execution_result`? (that function would not even need `&self` as well at that point.

Reply:
Original thought was to lock only when needed, and as short as possible. On the flip side, it increases number of locking significantly. probably not worth it. In any case, I agree it is better to keep locking behavior unchanged in this PR. Can bench and refactor separately.